compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -5349,6 +5349,30 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
         case UNWRAP_FROM_THROWABLE =>
           // js.special.unwrapFromThrowable(arg)
           js.UnwrapFromThrowable(genArgs1)
+
+        case LINKTIME_IF =>
+          // linkingInfo.linkTimeIf(cond, thenp, elsep)
+          assert(args.size == 3,
+              s"Expected exactly 3 arguments for JS primitive $code but got " +
+              s"${args.size} at $pos")
+          val condp = genLinkTimeCondition(args(0)).getOrElse {
+            // TODO better error reporting
+            reporter.error(tree.pos, s"Invalid")
```

Comment:
For better error messages, emit errors directly inside `genLinkTimeCondition` and have it return a non-`Option`.